File: userAPI/Mappers/UserMapper.cs
using userAPI.Domain.Users;

namespace userAPI.Mappers.Users {
    public class UserMapper {
        public static User toDomain(string nomeCompleto, string email, string pass, string telefone, string numMecanografico, string numeroContribuinte,  UserRole func) {
            return new User(nomeCompleto, email, pass, telefone, numMecanografico, numeroContribuinte, func);
        }

        public static UserDto toDTO(User user) {
            return new UserDto(user.Id.AsGuid(), user.nomeCompleto, user.email, user.password, user.telefone, user.numMecanografico, user.numeroContribuinte, user.func,user.approvalStatus);
        }
    }
}

File: frontend/src/app/mapper/UserMapper.ts
import UserDTO from "../dto/UserDTO";
import { UserModel } from "../model/userModel";

export class UserMapper {
    public static toUserModel(userDto: UserDTO): UserModel {

        return {
            id: userDto.Id,
            nomeCompleto: userDto.nomeCompleto,
            email: userDto.email,
            password: userDto.password,
            func: userDto.func,
            telefone: userDto.telefone,
            numMecanografico: userDto.numMecanografico,
            numeroContribuinte: userDto.numeroContribuinte

        };
    }

    public static toUserDTO(userModel: UserModel): UserDTO {
        return {
            Id: userModel.id,
            nomeCompleto: userModel.nomeCompleto,
            email: userModel.email,
            password: userModel.password,
            func: userModel.func,
            telefone: userModel.telefone,
            numMecanografico: userModel.numMecanografico,
            numeroContribuinte: userModel.numeroContribuinte

        };
    }
}
